db/schema.rb

```diff
@@ -1329,6 +1329,7 @@
     t.integer "purpose_id"
     t.integer "tube_rack_purpose_id"
     t.integer "rows_per_well"
+    t.text "invalid_wells", size: :medium
```

Comment:
The table is not a meta table to configure specific types of manifests. It is holding instances of manifest. It contains 9,575,207 records in production at the time of this writing. I don't think we need to record invalid wells for each manifest uploaded. The type of manifest should enforce it instead. It needs a specialised field and a validation on upload.

Reply:
This can give some ideas `app/sequencescape_excel/sequencescape_excel/specialised_field/bioscan_control_type.rb`

Reply:
It does feel like a shame / waste to have to add columns to the db for cases like these. I did the same thing with the 'rows_per_well' column, you can see above Ben's one. The problem is that the manifest upload code doesn't actually know what 'type' of manifest it's processing, so it can't pull these things from the manifest_type.yml configuration. It just has access to the saved manifest record. I thought this was a bit odd - it feels intuitive that it should be checking the config - but then I thought, maybe it's intentional. Maybe the reason was to keep it isolated from the config code in case you're re-uploading an old manifest and the manifest_types.yml config has changed since the manifest was generated? We could ask @stevieing in case he remembers the history of the design.

A specialised field feels wrong to me in this case - what would the column be called, and what would the users put in it?

Could have a face to face chat about this when Ben's back?

Reply:
Yeah happy to have a chat. As Katy mentioned we need some data persisted to verify the manifest isn't using the invalid wells (H12) on upload and we can't get that through the manifest type because we don't know what type it is.
I'll take a look at what the Bioscan one does as well 👍  Didnt realise we already did a similar thing.

Reply:
> It does feel like a shame / waste to have to add columns to the db for cases like these. I did the same thing with the 'rows_per_well' column, you can see above Ben's one. The problem is that the manifest upload code doesn't actually know what 'type' of manifest it's processing, so it can't pull these things from the manifest_type.yml configuration. It just has access to the saved manifest record. I thought this was a bit odd - it feels intuitive that it should be checking the config - but then I thought, maybe it's intentional. Maybe the reason was to keep it isolated from the config code in case you're re-uploading an old manifest and the manifest_types.yml config has changed since the manifest was generated? We could ask @stevieing in case he remembers the history of the design.
> 
> A specialised field feels wrong to me in this case - what would the column be called, and what would the users put in it?
> 
> Could have a face to face chat about this when Ben's back?

We created the configuration for creating the manifests and the processor for upload. One of the requirements at the time was that uploading needed to not be related to the manifest type. That was probably decided by James or Beth.